float64[3] fl_pos
float64[3] fr_pos
float64[3] rl_pos
float64[3] rr_pos
bool[4] contact_state